moto/dynamodb/models/__init__.py

```diff
@@ -1655,7 +1656,17 @@ def check_unicity(table_name, key):
             target_items.add(item)
 
         errors = []  # [(Code, Message, Item), ..]
+        ddb_verbs = ["ConditionCheck", "Put", "Delete", "Update"]
         for item in transact_items:
+            # check transact writes are not performing multiple operations
+            # in the same item
+            for ddb_verb in ddb_verbs:
```

Comment:
This logic doesn't seem quite right. It is possible to have different verbs appear, they just should not act on the same item.
An alternative solution that may work:
List the values in `transact_items`, so the items that are being PUT/DELETED/etc., and then verify there are no duplicates there.
The easiest solution to check for duplicates is `list(values) != set(values)` - the `set`-operation will eliminate any duplicates, so if the list contains more items - we know it's a dup.

Reply:
Ah, interesting. I completely misunderstood the issue, then. Thanks for the explanation.

Can we still use the `list(keys) == set(keys)`-approach, to verify there are no multiple verbs? That is more readable then iterating over the list twice, IMO

Reply:
Looks like our replies crossed - using `len(set(..))` would also work, yes :)